.github/workflows/build_rpms.yml

```diff
@@ -25,24 +25,44 @@ jobs:
           echo "fastestmirror=1" >> /etc/dnf/dnf.conf
           echo "install_weak_deps=0" >> /etc/dnf/dnf.conf
           dnf -y upgrade
-          dnf -y install dnf-plugins-core git make rpm-build
-          mkdir rpms
+          dnf -y install dnf-plugins-core rpm-build rpmdevtools
 
-      - name: Clone repository
-        uses: actions/checkout@v2
-        with:
-          path: osbuild
+      - name: Fetch .spec file
+        run: |
+          mkdir spec
+          # Fetch the spec file at the given commit, and prepend the commit sha to it
+          curl "https://raw.githubusercontent.com/osbuild/osbuild/${GITHUB_SHA}/osbuild.spec" -o osbuild.spec
+          echo "%global commit ${GITHUB_SHA}" | cat - osbuild.spec > spec/osbuild.spec
 
       - name: Install RPM build dependencies
-        run: dnf -y builddep osbuild/osbuild.spec
+        run: dnf -y builddep spec/osbuild.spec
+
+      - name: Fetch sources
+        run: |
+          mkdir sources
+          spectool -g spec/osbuild.spec -C sources/
+
+      - name: Build SRPM
+        run: |
+          mkdir srpms
+          rpmbuild -bs \
+            --define "_sourcedir sources" \
+            --define "_srcrpmdir srpms" \
+            spec/osbuild.spec
 
       - name: Build RPMs
         run: |
-          pushd osbuild
-            make rpm
-          popd
-          cp -av osbuild/output/*/*.rpm rpms/
-          date > rpms/timestamp.txt
+          mkdir rpmbuild
+          mkdir rpms
+          mkdir build
```

Comment:
You could use: `mkdir rpmbuild rpms build`
Not saying this is nicer, though.

Reply:
Will fix